client/Containers/FilterSortControl.jsx

```diff
@@ -0,0 +1,188 @@
+    ({ productFilters, productPagination }) => ({
+      productFilters,
+      productPagination,
+    })
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getProducts = (query) => dispatch(fetchProductList(query));
+  const filter = (options) => {
+    // options: { type?: string, style?: string, room?: string, clear?: true|false }
+    for (const key in productFilters) {
+      if (typeof options[key] === 'string') productFilters[key] = options[key];
+    }
+    if (options.clear) {
+      for (const key in productFilters) {
+        productFilters[key] = '';
+      }
+    }
+    const query = buildProductQuery({
+      productFilters,
+      productPagination: { ...productPagination, page: 1 },
+    });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+  const paginate = (options) => {
+    // options: { page?: INT, size?: INT, sort?: STRING }
+    for (const key in productPagination) {
+      if (options[key]) productPagination[key] = options[key];
+    }
+    const query = buildProductQuery({ productFilters, productPagination });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+  return (
+    <FilterSortBar>
+      <FilterContainer>
+        <FilterLabel>SORT</FilterLabel>
+        <FilterSelect
+          value={productPagination.sort}
+          onChange={(ev) => paginate({ page: 1, sort: ev.target.value })}
+        >
+          <option value={'name,ASC'}>Name, A-Z</option>
+          <option value={'name,DESC'}>Name, Z-A</option>
+          <option value={'price,DESC'}>Price, High to Low</option>
+          <option value={'price,ASC'}>Price, Low to High</option>
+        </FilterSelect>
+      </FilterContainer>
+      <FilterContainer>
+        <FilterLabel>SIZE</FilterLabel>
+        <FilterSelect
+          value={productPagination.size}
+          onChange={(ev) => paginate({ page: 1, size: ev.target.value * 1 })}
+        >
+          <option value="6">6 per Page</option>
+          <option value="12">12 per Page</option>
+          <option value="18">18 per Page</option>
+          <option value="24">24 per Page</option>
+        </FilterSelect>
+      </FilterContainer>
+      <AllFilterContainer>
+        {[
+          [
+            'room',
+            [
+              ['bedroom', 'Bedroom'],
+              ['living', 'Living'],
+              ['dining', 'Dining'],
+              ['bathroom', 'Bathroom'],
+              ['', 'None'],
+            ],
+          ],
+          [
+            'style',
+            [
+              ['contemporary', 'Contemporary'],
+              ['modern', 'Modern'],
+              ['transitional', 'Transitional'],
+              ['', 'None'],
+            ],
+          ],
+          [
+            'type',
+            [
+              ['bed', 'Beds'],
+              ['dresser', 'Dressers'],
+              ['nightstand', 'Nightstands'],
+              ['', 'None'],
+            ],
+          ],
```

Comment:
Probably want to get these options from a route at some point, but hard-coding them in here for now.